ops/_yaml.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Internal YAML helpers."""
+
+import yaml
+
+
+# Use C speedups if available
+_safe_loader = getattr(yaml, 'CSafeLoader', yaml.SafeLoader)
+_safe_dumper = getattr(yaml, 'CSafeDumper', yaml.SafeDumper)
+
+
+def safe_load(stream):
```

Comment:
Why are we not using yaml.safe_load and yaml.safe_dump themselves?

Can we do that and remove this logic and file entirely?

Reply:
That would be correct, but slow, because `safe_load` and `safe_dump` always use the slow Python versions of the loader, `SafeLoader`, whereas this uses `CSafeLoader` if it's available. See John's performance numbers above -- it makes quite a difference (not that performance is critical here). It's kinda daft that this logic isn't baked into PyYAML itself (maybe they didn't because CSafeLoader isn't quite backwards compatible with SafeLoader, I dunno?).

Reply:
For the record, we discussed this online and got to an agreement of keeping it as a file but moving it to a _private directory.